expression/util.go

```diff
@@ -503,3 +504,25 @@ func ColumnSliceIsIntersect(s1, s2 []*Column) bool {
 	}
 	return false
 }
+
+// DatumToConstant generates a Constant expression from a Datum.
+func DatumToConstant(d types.Datum, tp byte) *Constant {
```

Comment:
Can we put this function to types/datum.go since it's exported now?

Reply:
@XuHuaiyu 
I have a concern the cyclic dependency between the ``types`` package and the ``expression`` one because the function contains ``expression.Constant`` when the function is putted to ``types/datum.go``. Could you suggest another place?

Reply:
@dbjoa Ok, just keep it here. ^_^